vllm/config/profiler.py

```diff
@@ -0,0 +1,181 @@
+                "Environment variable VLLM_TORCH_CUDA_PROFILE is deprecated. "
+                "Please use profiler_config.profiler instead."
+            )
+        elif envs.VLLM_TORCH_PROFILER_DIR is not None:
+            self.profiler = "torch"
+            self.torch_profiler_dir = envs.VLLM_TORCH_PROFILER_DIR
+            logger.warning_once(
+                "Environment variable VLLM_TORCH_PROFILER_DIR is deprecated. "
+                "Please use profiler_config.torch_profiler_dir instead."
+            )
+
+            if envs.VLLM_TORCH_PROFILER_RECORD_SHAPES is not None:
+                self.torch_profiler_record_shapes = (
+                    envs.VLLM_TORCH_PROFILER_RECORD_SHAPES == "1"
+                )
+            if envs.VLLM_TORCH_PROFILER_WITH_PROFILE_MEMORY is not None:
+                self.torch_profiler_with_memory = (
+                    envs.VLLM_TORCH_PROFILER_WITH_PROFILE_MEMORY == "1"
+                )
+            if envs.VLLM_TORCH_PROFILER_WITH_STACK is not None:
+                self.torch_profiler_with_stack = (
+                    envs.VLLM_TORCH_PROFILER_WITH_STACK == "1"
+                )
+            if envs.VLLM_TORCH_PROFILER_WITH_FLOPS is not None:
+                self.torch_profiler_with_flops = (
+                    envs.VLLM_TORCH_PROFILER_WITH_FLOPS == "1"
+                )
+            if envs.VLLM_TORCH_PROFILER_DISABLE_ASYNC_LLM is not None:
+                self.ignore_frontend = envs.VLLM_TORCH_PROFILER_DISABLE_ASYNC_LLM == "1"
+            if envs.VLLM_TORCH_PROFILER_USE_GZIP is not None:
+                self.torch_profiler_use_gzip = envs.VLLM_TORCH_PROFILER_USE_GZIP == "1"
+            if envs.VLLM_TORCH_PROFILER_DUMP_CUDA_TIME_TOTAL is not None:
+                self.torch_profiler_dump_cuda_time_total = (
+                    envs.VLLM_TORCH_PROFILER_DUMP_CUDA_TIME_TOTAL == "1"
+                )
+
+        if envs.VLLM_PROFILER_DELAY_ITERS is not None:
+            self.delay_iterations = int(envs.VLLM_PROFILER_DELAY_ITERS)
+            logger.warning_once(
+                "Environment variable VLLM_PROFILER_DELAY_ITERS is deprecated. "
+                "Please use profiler_config.delay_iterations instead."
+            )
+        if envs.VLLM_PROFILER_MAX_ITERS is not None:
+            self.max_iterations = int(envs.VLLM_PROFILER_MAX_ITERS)
+            logger.warning_once(
+                "Environment variable VLLM_PROFILER_MAX_ITERS is deprecated. "
+                "Please use profiler_config.max_iterations instead."
+            )
+
+        if self.profiler not in ("none", "torch", "cuda"):
+            raise ValueError(
+                f"Invalid profiler: {self.profiler} "
+                f"(choose from 'none', 'torch', 'cuda')"
+            )
+        if self.delay_iterations < 0:
+            raise ValueError("Profiler delay_iterations must be >= 0")
+        if self.max_iterations < 0:
+            raise ValueError("Profiler max_iterations must be >= 0")
```

Comment:
- Pydantic will validate `profiler` for us
- For the other two you can default them to `Field(default=0, ge=0)`

```suggestion
```